src/source/catalog.ts

```diff
@@ -35,7 +35,7 @@ export class CatalogSource {
    * Creates an instance of CatalogSource.
    * @param {string} apikey An apiKey capable of pushing documents and managing sources in a Coveo organization. See [Manage API Keys](https://docs.coveo.com/en/1718).
    * @param {string} organizationid The Coveo Organization identifier.
-   * @param {PlatformUrlOptions} [opts=CatalogSource.defaultOptions]
+   * @param {PlatformUrlOptions} [opts=CatalogSource.defaultOptions] Platform request options.
```

Comment:
I don't think the `PlatformUrlOptions` interface is actually documented. 

You can verify that with vscode:

```typescript
new CatalogSource('api_key', 'org_id', {
   // Here, intellisense should pop up, and should show the doc
   // It can explain the default values, what it represent/what it's for exactly/some example of "good" values, etc.
   retryAfter: '....',
   // repeat for every new options
}
```

The comment that appears from Intellisense can be an exact copy paste of what would be on the README for each indiviual option.


Also, the name `PlatformUrlOptions` also does not make sense anymore. 
Just a generic `Options` would be better. It contains things that are unrelated to the URL

Make sure the same thing will also work for `src/source/push.ts`

Reply:
I've added JSDoc comments to the `PlatformOptions` type within `environment.ts`. I'm honestly not sure if there's a problem with Intellisense on my computer or something, because no amount changes seemed to have the kind of "intelligent" suggestions/descriptions while typing, so please let me know if this is what you were referring to.

I went with `PlatformOptions` since I found `Options` to be a bit _**too**_ open-ended.